app/api/server/v1/voip/omnichannel.ts

```diff
@@ -0,0 +1,94 @@
+import { Match, check } from 'meteor/check';
+
+import { API } from '../../api';
+import { Users } from '../../../../models/server/index';
+import { hasPermission } from '../../../../authorization/server/index';
+import { LivechatVoip } from '../../../../../server/sdk';
+
+API.v1.addRoute('omnichannel.agent.extension', { authRequired: true }, {
```

Comment:
```suggestion
API.v1.addRoute('omnichannel.agent.extension', { authRequired: true, permissionsRequired: ['query-agent-extension-association'] }, {
```

You don't have to explicity check for permission in your code. you can simply pass the permission you want on this route and then addRoute will take care of the rest

Reply:
Just realised that this comment won't be applicable to this route since each method required a different permission. So you can ingore this over here,

Reply:
Makes sense. No change as such on this. But will keep this in mind for other REST